examples/counter/apps/src/main.rs

```diff
@@ -81,10 +87,21 @@ async fn run(args: Args) -> Result<()> {
         .await
         .context("failed to build boundless client")?;
 
-    // We use a timestamp as input to the ECHO guest code as the Counter contract
-    // accepts only unique proofs. Using the same input twice would result in the same proof.
-    let echo_message = format!("{:?}", SystemTime::now());
-    let request = client.new_request().with_program(ECHO_ELF).with_stdin(echo_message.as_bytes());
+    // Build the request based on whether URLs are provided
+    let request = match (args.program_url.clone(), args.input_url.clone()) {
+        (Some(program_url), Some(input_url)) => {
+            // Use the provided URLs
+            client.new_request().with_program_url(program_url)?.with_input_url(input_url)?
+        }
```

Comment:
This part is not super necessary, I was using it to test the new `with_input_url`. I'm happy to revert the example back here.